File: src/main/java/um/haberes/core/hexagonal/personas/dependencia/domain/model/Dependencia.java
package um.haberes.core.hexagonal.personas.dependencia.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Dependencia {

    private Integer dependenciaId;

    @Builder.Default
    private String nombre = "";

    @Builder.Default
    private String acronimo = "";

    private Integer facultadId;

    private Integer geograficaId;

    public String getSedeKey() {
        return String.valueOf(facultadId) + "." + geograficaId;
    }
}

File: src/main/java/um/haberes/core/kotlin/model/Dependencia.kt
package um.haberes.core.kotlin.model

import jakarta.persistence.*
import um.haberes.core.hexagonal.facultad.infrastructure.persistence.entity.FacultadEntity
import um.haberes.core.hexagonal.geografica.infrastructure.persistence.entity.GeograficaEntity

@Entity
@Table
data class Dependencia(

    @Id
    var dependenciaId: Int? = null,
    var nombre: String = "",
    var acronimo: String = "",
    var facultadId: Int? = null,
    var geograficaId: Int? = null,

    @OneToOne(optional = false)
    @JoinColumn(name = "facultadId", insertable = false, updatable = false)
    var facultad: FacultadEntity? = null,

    @OneToOne(optional = false)
    @JoinColumn(name = "geograficaId", insertable = false, updatable = false)
    var geografica: GeograficaEntity? = null

) : Auditable() {

    fun getSedeKey(): String {
        return this.facultadId.toString() + "." + this.geograficaId
    }

}
